pallets/loans/src/rate.rs

```diff
@@ -41,13 +42,21 @@ impl From<Rate> for APR {
 impl APR {
     pub const MAX: Rate = Rate::from_inner(350_000_000_000_000_000); // 35%
 
+    pub fn from_fraction(delta_time: Timestamp) -> Option<Rate> {
+        FixedU128::from(1)
+            .checked_mul(&Rate::saturating_from_integer(delta_time))?
+            .checked_div(&Rate::saturating_from_integer(SECONDS_PER_YEAR))
+    }
+
     pub fn rate_per_block(&self, block_per_year: u128) -> Option<Rate> {
         self.0
             .checked_div(&Rate::saturating_from_integer(block_per_year))
     }
 
-    pub fn accrued_interest_per_block(&self, amount: u128, block_per_year: u128) -> Option<u128> {
-        self.0.checked_mul_int(amount)?.checked_div(block_per_year)
+    pub fn accrued_interest_per_block(&self, amount: u128, delta_time: Timestamp) -> Option<u128> {
+        let fraction: Rate = Self::from_fraction(delta_time)?;
+
+        fraction.checked_mul_int(self.0.checked_mul_int(amount)?)
```

Comment:
@yz89 is this ok now?

Reply:
`let fraction: Rate = Rate::checked_from_rational(delta_time, SECONDS_PER_YEAR);`

Get fraction directly.

Reply:
> @yz89 is this ok now?

yes